ocaml/xapi/xapi_cluster.ml

```diff
@@ -69,13 +69,21 @@ let create ~__context ~network ~cluster_stack ~pool_auto_join ~token_timeout ~to
 
 let destroy ~__context ~self =
   let dbg = Context.string_of_task __context in
-  assert_cluster_has_one_node ~__context ~self;
-  let cluster_host = Db.Cluster.get_cluster_hosts ~__context ~self |> List.hd in
-  assert_cluster_host_has_no_attached_sr_which_requires_cluster_stack ~__context ~self:cluster_host;
+  let cluster_hosts = Db.Cluster.get_cluster_hosts ~__context ~self in
+  let cluster_host = match List.length cluster_hosts with
+    | 0 -> None
+    | 1 -> Some (cluster_hosts |> List.hd)
```

Comment:
You can match the list directly and then you don't need to do List.hd here, i.e.
```
match cluster_hosts with
| [] -> None
| [ cluster_host ] -> Some cluster_host
| _ -> (* now you do List.length to report error *)
```
```